pkg/apis/infinispan/v1/types_util.go

```diff
@@ -248,6 +248,17 @@ func (ispn *Infinispan) NotClusterFormed(pods, replicas int) bool {
 	return notFormed || notEnoughMembers
 }
 
+func (ispn *Infinispan) EnsureClusterStability() error {
```

Comment:
I don't think `EnsureClusterStability` belongs in `type_util` as the logic is specific to the zero controller and it's perception of whether the cluster is stable enough for the zero pod to be created.

Reply:
@ryanemerson if you can see, I switched IsWellFormed function to use EnsureClusterStability as you implementation more correct for cluster validation than just check single condition

Reply:
O right, I missed that. In the case let's keep it. 

`IsWellFormed` is only used by the cache controller, maybe we should remove `IsWellFormed` and just update the call to:  
```go
if err := ispnInstance.EnsureClusterStability(); err != nil {
...
}
```

Reply:
I've found IsWellFormed in `autoscale` and test also. Are you agree to change all possible places with `EnsureClusterStability`. If yes, I will do.

Reply:
Sorry my search was on an outdated `master` .. if it has a few uses I don't mind keeping it, but whichever you prefer :+1: